### PR TITLE
Modify licenses tool to use "golang.org/x/tools/go/packages"

### PR DESCRIPTION
**Pull request by RJPercival**
Using this package, instead of "go/build", enables the tool to work with packages that are using Go Modules. It also transparently supports $GOFLAGS and handles walking the package dependency tree, which makes it possible to delete some code.

Contributes to #1400.

### Checklist

- [ ] I have updated the [CHANGELOG](CHANGELOG.md).
  - Adjust the draft version number according to [semantic versioning](https://semver.org/) rules.
- [x] I have updated [documentation](docs/) accordingly.

### CODE REVIEW

File: scripts/licenses/licenses/library.go [hunk 1]
```diff
@@ -15,78 +15,133 @@
...
+		}
 		if isStdLib(p) {
 			// No license requirements for the Go standard library.
-			continue
+			return false
+		}
+		var pkgDir string
+		switch {
+		case len(p.GoFiles) > 0:
+			pkgDir = filepath.Dir(p.GoFiles[0])
```
Comment (gdbelvin) [inline scripts/licenses/licenses/library.go]: do you want to concat all these options instead of terminating at the first case?

**Comment by gdbelvin** [comment]
Would you mind testing in a directory outside of GOPATH as well? 
I'm noticing that the original version of this tool fails when doing a fresh checkout with export GO111MODULE=on

**Comment by RJPercival** [comment]
I've tested it outside of $GOPATH and found that the `csv` subcommand is unable to determine URLs for the license files, because Go modules appear to have the `.git` directories deleted (which are necessary in order to determine where the repository is hosted). However, the `save` subcommand works, which is all we need for Travis to succeed.

**Comment by gdbelvin** [comment]
gosimple errors

**Comment by gdbelvin** [comment]
From the cgo docs:  https://golang.org/cmd/cgo/

When the Go tool sees that one or more Go files use the special import "C", it will look for other non-Go files in the directory and compile them as part of the Go package. Any .c, .s, or .S files will be compiled with the C compiler. Any .cc, .cpp, or .cxx files will be compiled with the C++ compiler. Any .f, .F, .for or .f90 files will be compiled with the fortran compiler. Any .h, .hh, .hpp, or .hxx files will not be compiled separately, but, if these header files are changed, the package (including its non-Go source files) will be recompiled. **Note that changes to files in other directories do not cause the package to be recompiled, so all non-Go source code for the package should be stored in the package directory, not in subdirectories.** The default C and C++ compilers may be changed by the CC and CXX environment variables, respectively; those environment variables may include command line options.

**Comment by RJPercival** [comment]
While that documentation doesn't preclude the C/C++/Fortran/etc. code from pulling in other dependencies, it seems like downgrading the existence of such code from an error to a warning isn't unreasonable. Perhaps I should add a user guide for this tool though so that people understand that this warning warrants investigation.

**Comment by RJPercival** [comment]
@gdbelvin: Could you take a look at the README and, if you're happy with it, I'll merge this PR.

**Comment by gdbelvin** [comment]
This looks good!

**Pull request merged**